Fix #371: add code generators

dminkovsky opened the pull request:
What do you think? Have never used Lombok before so didn't really have anything to say about it.

akullpp commented on README.md:
```diff
@@ -147,6 +147,14 @@ A curated list of awesome Java frameworks, libraries and software.
 
+## Code Generators
+
+*Code that generates bug-prone boilerplate code.*
```
`bug-prone`? That is rather harsh and opinionated.

dminkovsky replied:
Didn't mean to be harsh or opinionated! For context, that's just my understanding of code generation: If code is amenable to being generated, then it's probably the kind of code that is repetitive and therefore bug-prone.

But, that's just my take. Not looking to step on toes for that perspective. Please change it to whatever you want or let me know what it should say, if anything.

dminkovsky replied:
To be clear, I was not talking about the code that the code generators
generate. "bug-prone" was the code you would otherwise have to write
yourself, by hand.

On Fri, Jul 1, 2016 at 4:05 AM, Andreas Kull <notifications@github.com>
wrote:

> In README.md
> <https://github.com/akullpp/awesome-java/pull/375#discussion_r69263432>:
>
> > @@ -147,6 +147,14 @@ A curated list of awesome Java frameworks, libraries and software.
> >  * [Cobertura](http://cobertura.github.io/cobertura/) - Relies on offline (or static) bytecode instrumentation and class loading to collect code coverage metrics; GPLv2 licensed.
> >  * [JaCoCo](http://eclemma.org/jacoco/) - Framework that enables collection of code coverage metrics, using both offline and runtime bytecode instrumentation; prominently used by EclEmma, the Eclipse code-coverage plugin.
> >
> > +## Code Generators
> > +
> > +*Code that generates bug-prone boilerplate code.*
>
> bug-prone? That is rather harsh and opinionated.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/akullpp/awesome-java/pull/375/files/a4eafb68cbac8895c1c4467fc5f862cb96effd1e#r69263432>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AANWZd3N801R7WS1YP2bdG1sYPYd5gwuks5qRMpfgaJpZM4I_Khm>
> .
>

akullpp replied:
Then could you please remove the `bug-prone`?

akullpp commented on README.md:
```diff
@@ -147,6 +147,14 @@ A curated list of awesome Java frameworks, libraries and software.
+
+*Code that generates bug-prone boilerplate code.*
+
+* [FreeBuilder](https://github.com/google/FreeBuilder) - Automatic generation of the Builder pattern for Java 1.6+ with Java 8 support.
```
Please remove ` for Java 1.6+ with Java 8 support`

akullpp commented on README.md:
```diff
@@ -147,6 +147,14 @@ A curated list of awesome Java frameworks, libraries and software.
+*Code that generates bug-prone boilerplate code.*
+
+* [FreeBuilder](https://github.com/google/FreeBuilder) - Automatic generation of the Builder pattern for Java 1.6+ with Java 8 support.
+* [Auto](https://github.com/google/auto) - Generates factory, service, and value classes and provides other utils.
```
Specify `other` or leave it out.

akullpp commented on README.md:
```diff
@@ -147,6 +147,14 @@ A curated list of awesome Java frameworks, libraries and software.
+
+* [FreeBuilder](https://github.com/google/FreeBuilder) - Automatic generation of the Builder pattern for Java 1.6+ with Java 8 support.
+* [Auto](https://github.com/google/auto) - Generates factory, service, and value classes and provides other utils.
+* [Lombok](https://github.com/rzwitserloot/lombok) - Generates Java boilerplate.
```
We have Lombok already added. Therefore you should move the entry:

`Lombok - Code-generator which aims to reduce the verbosity.` 

out of `Development`.

dminkovsky replied:
Missed this comment before. Will do.

dminkovsky commented:
Just saw your other comments. I think I've addressed all of them. 

Also moved Immutables and used the lead sentence from the library's webpage as the new description.

jbduncan commented:
One other little suggestion from me - in Lombok's description, it should say "code generator" rather than "code-generator", to be consistent with the heading "Code Generators". :smiley_cat:

dminkovsky commented:
Ah thanks. Fixed that.

четверг, 14 июля 2016 г. пользователь Jonathan Bluett-Duncan написал:

> One other little suggestion from me - in Lombok's description, it should
> say "code generator" rather than "code-generator", to be consistent with
> the heading "Code Generators". 😺
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/akullpp/awesome-java/pull/375#issuecomment-232723415>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AANWZR1GAjSvjIljKUkb9uI_Rewvb7Efks5qVmhigaJpZM4I_Khm>
> .
>

akullpp commented:
Please add the new section to the Table of Contents. @dminkovsky

jbduncan commented on README.md:
```diff
@@ -258,10 +267,8 @@ A curated list of awesome Java frameworks, libraries and software.
 * [Auto](https://github.com/google/auto) - Collection of source code generators.
```
Oh, I just noticed that `Auto` exists here too. I think it should be removed here. (The entry under Code Generators can stay.)
